File: cli/src/modules/dockerfile/sqlite.ts
import type { DockerfileModule } from '@/core/types.js';

export const sqliteModule: DockerfileModule = {
  id: 'sqlite',
  label: 'SQLite',
  category: 'db',
  render() {
    return `##
## SQLITE
##
RUN apt-get update && apt-get install -y sqlite3
`;
  },
};

File: cli/internal/domain/modules/dockerfile/sqlite.go
package dockerfile

import (
	"fmt"

	"github.com/joacohbc/my-devcontainer-installer/cli/internal/domain/types"
)

var SqliteModule = &ModuleSpec{
	ID:         types.ModuleSqlite,
	Label:      "SQLite",
	Category:   types.CategoryDB,
	UICategory: types.UICategoryDatabases,
	Context: func(opts map[string]any) *types.ContextSection {
		return &types.ContextSection{
			Title: "SQLite",
			Body: ctxBody(
				"`sqlite3` is installed. A database file under `/workspace` survives the",
				"container; one written anywhere else does not.",
			),
		}
	},
	Render: func(opts map[string]any) string {
		return fmt.Sprintf(`##
## SQLITE
##
RUN apt-get update && apt-get install -y sqlite3 && %s
`, aptCleanup())
	},
}
